arbitrum-docs/launch-orbit-chain/how-tos/customize-stf.mdx

```diff
@@ -0,0 +1,217 @@
+---
+title: "How to customize your Orbit chain's behavior"
+sidebar_label: "Customize your chain's behavior"
+description: "Learn how to customize your Orbit chain's behavior (known as the State Transition Function or STF)"
+author: plasmapower
+sidebar_position: 2
+---
+
+import PublicPreviewBannerPartial from '../partials/_orbit-public-preview-banner-partial.md';
+
+<PublicPreviewBannerPartial />
+
+# Preface
+
+Before customizing your Orbit chain, it's important to understand what the State Transition Function (aka the STF) is.
+The STF defines how new blocks are produced from input messages (i.e. transactions).
+This guide is only necessary for changes that modify the State Transition Function.
+To customize other node behavior, such as RPC behavior or the sequencer's ordering policy, you can simply
+[build your own node](/node-running/how-tos/build-nitro-locally) without worrying about the rest of this guide.
+However, changes that modify the STF require updating the fraud proving system to recognize the new behavior as correct.
+Otherwise, the fraud prover would side with unmodified nodes, which would win fraud proofs against your modified node.
+
+Here's some examples of modifications that affect the STF:
+
+- Adding a new EVM opcode or precompile:
+  This modifies the STF because a node with this change would disagree about the outcome of EVM execution compared to an unmodified nitro node
+  when the new opcode or precompile is invoked.
+- Rewarding the deployer of a smart contract with a portion of gas spent in the smart contract's execution:
+  This modifies the STF because a node which has this change applied would disagree about the balance of the deployer after transactions. Such changes would lead to disagreements about block hashes compared to unmodified Nitro nodes.
+
+Here's some examples of modifications that don't affect the STF:
+
+- Adding a new RPC method to query an address's balance across multiple blocks:
+  This doesn't modify the STF because this doesn't change on-chain balances or block hashes.
+- Changing the sequencer to order blocks by tip:
+  The sequencer is trusted to order transactions in Arbitrum Nitro, and it can chose any ordering it wants.
+  Nodes (and the fraud proofs) will simply accept the new transaction ordering as there is no single ordering they think is correct.
+
+## Modification Compatibility with Arbitrum Nitro
+
+Some potential modifications are incompatible with Arbitrum Nitro and would not result in a functioning blockchain.
+Here are some requirements for the Arbitrum Nitro State Transition Function:
+
+- The STF must be deterministic. For instance, if you gave an address a random balance using the Go randomness library,
+  every node would disagree on the correct amount of balance and the blockchain would not function correctly.
+  However, it is acceptable to take a non-deterministic path to a deterministic output.
```

Comment:
Pedantic point, but this made me think of the non-deterministic preimages issue caused by iteration over a map in the go-ethereum code that we solved :D